### PR TITLE
New version: Waterfox.Waterfox version G6.0.5

### PR DESCRIPTION
**Pull request by aaronliu0130**
###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/124730)

### CODE REVIEW

**Comment by aaronliu0130** [comment]
image
uhhh

**Comment by stephengillie** [comment]
> image uhhh

This seems like a continuation from [here](https://github.com/microsoft/winget-pkgs/pull/122324#issuecomment-1792420539):
> I checked the G6.0.5 release, unfortunately it has the same DisplayVersion 115.4.1. This means we'll just have to remove G6.0.3 (skip 0.4) and add 0.5

Is there any other version currently in the manifest that has this same entry?

**Comment by aaronliu0130** [comment]
No.

**Comment by aaronliu0130** [comment]
@mdanish-kh, any thoughts?

**Comment by mdanish-kh** [comment]
@aaronliu0130 G6.0.3 in fact does exist currently in the repo. As i stated in an earlier comment, https://github.com/microsoft/winget-pkgs/pull/122324#issuecomment-1792381064, it appears we need to handle removal in a separate PR

**Comment by mdanish-kh** [comment]
> No. At least there shouldn’t be. I’ve deleted .03.

This deletion only exists currently in "your" branch, not in the master repo of microsoft/winget-pkgs. (well until we raise a PR for removal that gets merged in)

**Comment by mdanish-kh** [comment]
I've opened up a PR to unblock this one
- https://github.com/microsoft/winget-pkgs/pull/124991

**Comment by aaronliu0130** [comment]
> 2023-11-06 16:37:54.875 [CLI ] Installer [X64,nullsoft,Machine,] not applicable: Installer scope does not match required scope: Machine != User

but the redist dependency was removed already, why

**Review by mdanish-kh — CHANGES_REQUESTED** [review]
By my testing, the installer only installs in `machine` scope. Elevation doesn't seem to affect the scope in which the installer is installed in. Are there any silent switches to control what scope is the app being installed in?  If not, then we need to only include `Scope: machine`

**Comment by mdanish-kh** [comment]
Looks like `Manifest-AppsAndFeaturesVersionError` is gone now 🎉 Now we need @stephengillie to apply a waiver for `Validation-Installation-Error` if applicable after you commit the suggestions above.

> /AzurePipelines run

(Just a note: @stephengillie  the label `Manifest-AppsAndFeaturesVersionError` was re-applied when you ran the pipelines above even tho the removal PR had gotten merged is because the changes didn't make their way to the index (source.msix) file yet. After the PR passed publish pipeline, changes got added in the index and `Manifest-AppsAndFeaturesVersionError` didn't get re-applied)

**Comment by aaronliu0130** [comment]
No, it does install in user scope. Just press "no" in the UAC pop-up. This is also the case for Firefox, Thunderbird etc.'s offline installers.

File: manifests/w/Waterfox/Waterfox/G6.0.5/Waterfox.Waterfox.installer.yaml
Comment (mdanish-kh) [inline manifests/w/Waterfox/Waterfox/G6.0.5/Waterfox.Waterfox.installer.yaml:9]:
```suggestion
InstallerType: nullsoft
ElevationRequirement: elevatesSelf
```

File: manifests/w/Waterfox/Waterfox/G6.0.5/Waterfox.Waterfox.installer.yaml
Comment (mdanish-kh) [inline manifests/w/Waterfox/Waterfox/G6.0.5/Waterfox.Waterfox.installer.yaml:33]:
```suggestion
```

File: manifests/w/Waterfox/Waterfox/G6.0.5/Waterfox.Waterfox.installer.yaml
Comment (mdanish-kh) [inline manifests/w/Waterfox/Waterfox/G6.0.5/Waterfox.Waterfox.installer.yaml:38]:
```suggestion
```

**Review by mdanish-kh — CHANGES_REQUESTED** [review]
Ah, so this is similar to the `Git.Git` package which handles scope in a similar way. WinGet currently doesn't handle these scenarios properly. See

- https://github.com/microsoft/winget-cli/issues/3240

We should follow how [Git's manifest](https://github.com/microsoft/winget-pkgs/blob/master/manifests/g/Git/Git/2.42.0/Git.Git.installer.yaml) is setup, wherein `elevatesSelf` property is defined at the root (applicable for both scopes). This property is defined because the installer does "request" for elevation, it's just up to the user to accept/deny and be well-aware of the consequences of both

**Review by mdanish-kh — APPROVED** [review]

**Pull request merged**
